File: problems/p_12/p12.cpp
/*
project euler problem 12
*/
#include <iostream>

int divisorCount(long num){
  int count = 2;
  int i = 2;
  while (i * i < num) {
    if (num % i == 0) {
      count += 2;
    }
    i++;
  }
  return count;
}

long solution(){
  long number = 1;
  long ordinal = 2;
  while (divisorCount(number) < 500){
    number += ordinal;
    ordinal++;
  }
  return number;
}

int main(int argc, char const *argv[]) {
  std::cout << solution() << '\n';
  return 0;
}

File: 12/p12.py
"""
project euler problem 12
"""

def divisor_count(num):
    count = 2
    i = 2
    while i * i < num:
        if num % i == 0:
            count += 2
        i += 1
    return count

def solution():
  number = 1
  ordinal = 2
  while divisor_count(number) < 500:
      number += ordinal
      ordinal += 1
  return number


def main():
    print(solution())

if __name__ == '__main__':
    main()
